src/test/java/unit/SpendProfileUser.java

```diff
@@ -0,0 +1,197 @@
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.ClearContextUnit;
+import harness.utils.SamGroups;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra spend` commands. */
+@Tag("unit")
+public class SpendProfileUser extends ClearContextUnit {
+  // only an owner on the spend profile can disable emails
+  TestUsers spendProfileOwner = TestUsers.chooseTestUserWithOwnerAccess();
+
+  SamGroups trackedGroups = new SamGroups();
+
+  @AfterAll
+  void cleanupOnce() throws IOException, InterruptedException {
+    spendProfileOwner.login();
+
+    // try to disable spend for each group that was created by a method in this class
+    for (Map.Entry<String, TestUsers> groupNameToCreator :
+        trackedGroups.getTrackedGroups().entrySet()) {
+      UFGroup groupInfo =
+          TestCommand.runAndParseCommandExpectSuccess(
+              UFGroup.class, "groups", "describe", "--name=" + groupNameToCreator.getKey());
+      TestCommand.runCommand("spend", "disable", "--email=" + groupInfo.email, "--policy=USER");
+      TestCommand.runCommand("spend", "disable", "--email=" + groupInfo.email, "--policy=OWNER");
+    }
+
+    // then try to delete the groups
+    trackedGroups.deleteAllTrackedGroups();
+  }
+
+  @Test
+  @DisplayName("list-users reflects enabling and disabling users")
+  void listUsersReflectsEnableDisable() throws IOException {
+    // get an email to try and add
+    String testEmail = generateSamGroupForEmail();
+
+    // use a test user that is an owner to grant spend access to the test email
+    TestUsers spendProfileOwner = TestUsers.chooseTestUserWithOwnerAccess();
+    spendProfileOwner.login();
+
+    // `terra spend enable --email=$testEmail --policy=USER`
+    TestCommand.runCommandExpectSuccess("spend", "enable", "--email=" + testEmail, "--policy=USER");
+
+    // check that the test email is included in the list-users output
+    expectListedUserWithPolicies(testEmail, SpendProfilePolicy.USER);
+
+    // `terra spend disable --email=$testEmail --policy=USER`
+    TestCommand.runCommandExpectSuccess(
+        "spend", "disable", "--email=" + testEmail, "--policy=USER");
+
+    // check that the test email is not included in the list-users output
+    Optional<UFSpendProfileUser> listUsersOutput = listUsersWithEmail(testEmail);
+    assertTrue(
+        listUsersOutput.isEmpty(),
+        "user is not included in the test-users output after having their spend access disabled");
+  }
+
+  @Test
+  @DisplayName("only users who are an admin on the spend profile can enable and disable others")
+  void onlyAdminCanEnableDisable() throws IOException {
```

Comment:
Is this testing CLI functionality or Sam functionality?

Reply:
I guess it's hard to separate the two entirely. I wrote this test thinking about whether the CLI test users were setup correctly. For example, if a user is an owner on the spend profile, when they're supposed to be a user, then the tests that expect spend profile access would still work, but we're not testing what we think we are. I thought this was important to test, because I had trouble with this when I tested briefly against an early Verily SAM (i.e. users that were direct owners on the spend profile worked, but those that were users or members of a group that was a user did not).